content/blog/analysis-with-RCrunch.Rmd

```diff
@@ -0,0 +1,300 @@
+```{r}
+all_remote <- so_data[so_data$HomeRemote == "All or almost all the time (I'm full-time remote)", ]
+some_remote <- so_data[so_data$HomeRemote != "All or almost all the time (I'm full-time remote)" & so_data$HomeRemote != "Never", ]
+never_remote <- so_data[so_data$HomeRemote == "Never", ]
+```
+```{r}
+mean(all_remote$JobSatisfaction, na.rm=TRUE)
+mean(some_remote$JobSatisfaction, na.rm=TRUE)
+mean(never_remote$JobSatisfaction, na.rm=TRUE)
+```
+
+Let's see how these two approaches compare time-wise.
+```{r}
+clearCache() # let's make this a fair fight
+system.time({
+    remote_col <- as.vector(so_data$HomeRemote)
+    is_all_remote <- remote_col == "All or almost all the time (I'm full-time remote)"
+    is_some_remote <- remote_col != "All or almost all the time (I'm full-time remote)" & remote_col != "Never"
+    is_never_remote <- remote_col == "Never"
+    
+    job_satisfaction <- as.vector(so_data$JobSatisfaction)
+    mean(job_satisfaction[is_all_remote], na.rm=TRUE)
+    mean(job_satisfaction[is_some_remote], na.rm=TRUE)
+    mean(job_satisfaction[is_never_remote], na.rm=TRUE)
+})
+```
+
+```{r}
+clearCache() # let's make this a fair fight
+system.time({
+    all_remote <- so_data[so_data$HomeRemote == "All or almost all the time (I'm full-time remote)", ]
+    never_remote <- so_data[so_data$HomeRemote == "Never", ]
+    some_remote <- so_data[so_data$HomeRemote != "All or almost all the time (I'm full-time remote)" & so_data$HomeRemote != "Never", ]
+    
+    mean(all_remote$JobSatisfaction, na.rm=TRUE)
+    mean(some_remote$JobSatisfaction, na.rm=TRUE)
+    mean(never_remote$JobSatisfaction, na.rm=TRUE)
+})
+```
+Again, using the Crunch package's methods are faster. But not only that, we don't have to worry about the information in each column staying in the same order. With the download then process approach, if we reordered the `JobSatisfaction` column, our `is_all_remote` mask would no longer correspond to the correct values, and our means would be totally wrong[^fn1]. Using Crunch filters and the mean calculation provided by the Crunch package is immune to this, because the values are connected within rows on the server.
+
+## Cross tabulations
+We've already started looking at how `JobSatisfaction` varies by `HomeRemote` status using a fairly course grouping by filtering the rows to use in each mean. We can approach this in a more systematic way by using cross tabulations. Cross tabulations (also known as crosstabs or contingency tables) is one way to show the relationship between two variables. With Crunch, cross tabulations are incredibly easy (in the web app, it's literally drag and drop), and the best part is that we still don't have to transfer lots of data from Crunch.
```

Comment:
> And let’s face it: crosstabs make up the majority of analyses and deliverables when conducting survey research. 

Also maybe note that crtabs is not just about crosstabs, but is a namespaced `xtabs`. I don’t think I learned about `xtabs` for *years* and definitely have introduced it to others who should have known about it.

Also, maybe gesture at (or maybe not) that you can do nway crostabs?
> Does this relationship hold across countries? We have another categorical variable, country, and can just `+ country` …